sebsync.py

```diff
@@ -396,17 +497,103 @@ def sebsync(**kwargs):
         if download_new:
             path = options.downloads / ebook_filename(remote_ebook)
             download_ebook(remote_ebook.href, path, Status.NEW)
+            if options.type == "kindle":
+                hexdigest = calculate_hash(path)
+                download_already_exists = False
+                for title in local_cache:
+                    # If the hexdigest of the downloaded file matches an existing entry,
+                    # then we've seen this book before
+                    if local_cache[title].get("hexdigest") == hexdigest:
+                        cache_entry_path = local_cache[title].get("path")
+
+                        # Check to see if the book we've seen before is still present
+                        if (
+                            cache_entry_path
+                            and cache_entry_path.exists()
+                            and cache_entry_path.is_file()
+                        ):
+                            click.echo(f"{path.name} is the same as existing file {title}!")
+                            download_already_exists = True
+                if not options.dry_run:
+                    # If the book we saw before is still present, remove it
+                    if download_already_exists:
+                        path.unlink()
+
+                    # Otherwise, add it to our local index
+                    else:
+                        local_cache[path.name] = {
+                            "id": remote_ebook.id,
+                            "title": remote_ebook.title,
+                            "modified": remote_ebook.updated,
+                            "hexdigest": hexdigest,
+                        }
 
     for local_ebook in local_ebooks:
         if local_ebook.id not in remote_ebooks:
             if is_deprecated(local_ebook):
                 if options.remove:
                     remove(local_ebook)
+                    if options.type == "kindle" and not options.dry_run:
+                        local_cache.pop(local_ebook.path.name, None)
                 else:
                     echo_status(local_ebook.path, Status.OUTDATED)
             else:
                 echo_status(local_ebook.path, Status.EXTRA)
 
+    if options.type == "kindle":
+        # Remove titles from the cache if they don't exist locally;
+        # first create a list of all the local titles
+        local_titles = {}
+
+        if options.type == "kindle":
```

Comment:
Isn't it guaranteed to be kindle already from line 543 above?

Reply:
Yup; removed via https://github.com/gfitzp/sebsync/commit/6c35a156671a9189ea582e8351f3925bc50e7997#diff-c0aa284251fd36185582827235e0310f6707723753235a2a9c3fcf98f7d43497L544